lib/cfe_internal_hub.cf

```diff
@@ -216,7 +216,7 @@ bundle agent cfe_internal_database_cleanup_consumer_status (row_count)
     any::
 
       "remove_query"
-        string => "DELETE FROM $(status_table_name) WHERE ts IN (SELECT ts FROM $(status_table_name) ORDER BY ts DESC OFFSET $(row_count));";
+        string => "SELECT * FROM cleanup_historical_data('$(status_table_name)', 'ts', $(row_count), 'host');";
```

Comment:
Since the policy is expected to run on older versions where the function is not available we need to set the query appropriately. I can think of 2 ways to check for it. 1) grep the schema.sql for the function (weak), 2) query psql directly (better?)

```
  bundle agent main
  {
    classes:
      "has_sql_function_cleanup_historical_data"
       expression => regline( "^CREATE OR REPLACE FUNCTION cleanup_historical_data($",
                              "/var/cfengine/share/db/schema.sql" );

      "has_sql_function_cleanup_historical_data"
        expression => returnszero( "$(sys.bindir)/psql cfdb -c \"SELECT 'cleanup_historical_data'::regproc;", noshell );


    vars:
        "status_table_name" string => "status";
        "row_count" string => "3";
        "query" string => "DELETE FROM $(status_table_name) WHERE ts IN (SELECT ts FROM $(status_table_name) ORDER BY ts DESC OFFSET $(row_count));"

      has_sql_function_cleanup_historical_data::
        "query" string => "SELECT * FROM cleanup_historical_data('$(status_table_name)', 'ts', $(row_count), 'host'";
  }

```

Reply:
https://github.com/aleksandrychev/masterfiles/pull/1